### PR TITLE
Updates Refine for Gene Update

### PR DESCRIPTION
**Pull request by orta**
https://github.com/artsy/eigen/issues/1293

I've removed all the GeneViewModel bits, as that's all done in Relay now. Also spotted that we had duplicated CocoaPods Scripts in the build phases, I wonder if we should add a danger warning for that, I guess someone's occasionally not running with `bundle install` - will comment inline

### CODE REVIEW

File: Artsy/View_Controllers/Auction/AuctionRefineSettings.swift [hunk 1]
```diff
@@ -15,6 +15,10 @@ extension AuctionRefineSettings: RefinableType {
         return 1
     }
 
+    func titleOfSection(section: Int) -> String {
+        return "Sort"
+    }
```
Comment (orta) [inline Artsy/View_Controllers/Auction/AuctionRefineSettings.swift]: Needed this as it was implicit in the RefineVC previously

File: Artsy/View_Controllers/Gene/GeneRefineSettings.swift [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+    let mediums: [String]
+    var priceRange: PriceRange?
+
+    static func refinementFromJSON(data: [String: AnyObject]) -> GeneRefineSettings {
+        return GeneRefineSettings(sort: .RecentlyAdded, medium:"ok", mediums:["ok", "sure"], priceRange: (min:0, max: 50_000))
+    }
+
+    func toJSON() -> [String: AnyObject] {
+        return [:]
+    }
```
Comment (orta) [inline Artsy/View_Controllers/Gene/GeneRefineSettings.swift]: All this to <-> from JSON stuff will be coming in an upcoming PR - its out of scope for getting the refine up to the new specs

File: Artsy/View_Controllers/Util/ARScrollNavigationChief.h [hunk 3]
```diff
@@ -1,6 +1,6 @@
 /// Psh, managers are for Java
 
-#import "ARExternalWebBrowserViewController.h"
+#import <UIKit/UIkit.h>
```
Comment (orta) [inline Artsy/View_Controllers/Util/ARScrollNavigationChief.h]: I assume someone thought this was an implicit dependency, but I couldn't find it, so I switched it to `UIKit`

File: Artsy/View_Controllers/Util/ARScrollNavigationChief.m [hunk 4]
```diff
@@ -2,8 +2,6 @@
 
 #import "ARMacros.h"
 
-#import <ReactiveCocoa/ReactiveCocoa.h>
```
Comment (orta) [inline Artsy/View_Controllers/Util/ARScrollNavigationChief.m]: I assume this was used for our `ARMacros` - we don't use it here

File: Artsy_Tests/App_Tests/ARSwitchBoardTests.m [hunk 5]
```diff
@@ -328,15 +328,15 @@ @interface ARTopMenuNavigationDataSource (Tests)
...
-        });
+//        it(@"routes modern genes when using the dev option", ^{
+//            BOOL originalModernGene = [AROptions boolForOption:AROptionsUseModernGeneVC];
+//            [AROptions setBool:YES forOption:AROptionsUseModernGeneVC];
+//
+//            id subject = [switchboard routeInternalURL:[NSURL URLWithString:@"http://artsy.net/gene/surrealism"] fair:nil];
+//            expect(subject).to.beAKindOf(ModernGeneViewController.class);
+//
+//            [AROptions setBool:originalModernGene forOption:AROptionsUseModernGeneVC];
+//        });
```
Comment (orta) [inline Artsy_Tests/App_Tests/ARSwitchBoardTests.m]: This will come back with a component based VC

File: Artsy_Tests/View_Controller_Tests/Auction/RefinementOptionsViewControllerTests.swift [hunk 6]
```diff
@@ -56,6 +56,14 @@ class RefinementOptionsViewControllerSpec: QuickSpec {
             expect(subject).to( haveValidSnapshot() )
```
Comment (orta) [inline Artsy_Tests/View_Controller_Tests/Auction/RefinementOptionsViewControllerTests.swift]: ^ This scrolls

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          3 Errors
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">Release 2.6.4 does not have a date</td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="true">There were <a href="https://eigen-ci.s3.amazonaws.com/3321/index.html">snapshot errors</a></td>
    </tr>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>

    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Needs testing on a Phone if change is non-trivial</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="true">Detected some Swift building time outliers</td>
    </tr>
  </tbody>
</table>

Time | Class | Function |
| --- | ----- | ----- |286.4ms | [AuctionViewController.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Auction/AuctionViewController.swift#L199) | @objc dynamic func showRefineTappedAnimated(animated: Bool)
270.9ms | [SwiftyJSON.swift](/artsy/eigen/tree/orta-gene_refine/Pods/SwiftyJSON/Source/SwiftyJSON.swift#L445) | get {}
206.1ms | [WebSocket.swift](/artsy/eigen/tree/orta-gene_refine/Pods/Starscream/Source/WebSocket.swift#L528) | private func processRawMessage(buffer: UnsafePointer<UInt8>, bufferLen: Int)
82.1ms | [Observable.swift](/artsy/eigen/tree/orta-gene_refine/Pods/Interstellar/Sources/Observable.swift#L71) | private final func nextTokenHash() -> Int
80.9ms | [AuctionLotMetadataStackScrollView.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Live_Auctions/Views/AuctionLotMetadataStackScrollView.swift#L18) | required init(viewModel: LiveAuctionLotViewModelType, salesPerson: LiveAuctionsSalesPersonType, sideMargin: String)
74.3ms | [LiveAuctionsAdminViewController.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Live_Auctions/LiveAuctionsAdminViewController.swift#L22) | @objc override func viewDidLoad()
69.5ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L25) | func ar_presentBlurredOverlayWithTitle(title: String, subtitle: String, buttonState: BlurredStatusOverlayViewCloseButtonState = default)
69.4ms | [LiveAuctionLotViewController.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Live_Auctions/LiveAuctionLotViewController.swift#L49) | @objc override func viewDidLoad()
67.2ms | [UIViewController+BlurredStatusView.swift](/artsy/eigen/tree/orta-gene_refine/Artsy/View_Controllers/Core/UIViewController+BlurredStatusView.swift#L31) | (closure)


### Tests: 

Classname | Name|
--- | ---|
ARArtistViewControllerSpec | test_as_iphone|
ARCreateAccountViewControllerSpec | test_warning_view__displays_a_message|
ARNavigationControllerSpec | test_visuals__should_be_empty_for_a_rootVC|
ARNavigationControllerSpec | test_visuals__should_be_show_a_back_button_with_2_view_controllers|
ARSwitchBoardSpec | test_ARSwitchboard__routeInternalURL__routes_artists|
ARSwitchBoardSpec | test_ARSwitchboard__routeInternalURL__routes_artists_in_a_gallery_context_on_iPad|
ARSwitchBoardSpec | test_ARSwitchboard__routeInternalURL__routes_live_auctions|
ARTabViewSpec | test_sets_current_and_previous_view_index|
ARTabViewSpec | test_correctly_shows_a_navigation_controller__at_first_index|
ARTabViewSpec | test_correctly_shows_a_navigation_controller__at_another_index|
ARTabViewSpec | test_correctly_sets_the_child_view_controller|
ARTabViewSpec | test_sends_appearance_message_to_child_view_controller__viewDidLoad|
ARTabViewSpec | test_sends_appearance_message_to_child_view_controller__viewWillAppear|
ARTabViewSpec | test_sends_appearance_message_to_child_view_controller__viewDidAppear|
ARTabViewSpec | test_sends_appearance_message_to_child_view_controller__viewWillDisappear|
ARTabViewSpec | test_sends_appearance_message_to_child_view_controller__viewDidDisappear|
ARTopMenuNavigationDataSourceSpec | test_uses_a_single_feed_vc|
ARTopMenuNavigationDataSourceSpec | test_uses_a_single_browse_vc|
ARTopMenuNavigationDataSourceSpec | test_reinstantiates_favorites_vc|
ARTopMenuNavigationDataSourceSpec | test_notifications__sets_the_app_icon_badge_to_the_total_amount_of_available_notifications|
ARTopMenuNavigationDataSourceSpec | test_notifications__resets_the_app_icon_badge_if_there_are_0_notifications|
ARTopMenuViewControllerSpec | test_as_iphone|
ARTopMenuViewControllerSpec | test_as_ipad|
ARTopMenuViewControllerSpec | test_as_iphone_2|
ARTopMenuViewControllerSpec | test_as_ipad_2|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__is_selectable_when_not_selected|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__already_selected__is_not_selectable|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__already_selected__pops_to_root|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__animates_popping|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__does_not_change_tab|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__when_presenting_a_root_view_controller__when_not_on_the_selected_tab__changes_tabs_in_an_animated_fashion|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__concerning_badges__shows_a_notification_badge|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__concerning_badges__updates_the_badge_count_in_the_data_source|
ARTopMenuViewControllerSpec | test_navigation__feed__tab_behavior__concerning_badges__does_not_show_a_notification_badge_when_its_value_is_0|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__is_selectable_when_not_selected|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__already_selected__is_not_selectable|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__already_selected__pops_to_root|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__animates_popping|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__does_not_change_tab|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__when_presenting_a_root_view_controller__when_not_on_the_selected_tab__changes_tabs_in_an_animated_fashion|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__concerning_badges__shows_a_notification_badge|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__concerning_badges__updates_the_badge_count_in_the_data_source|
ARTopMenuViewControllerSpec | test_navigation__browse__tab_behavior__concerning_badges__does_not_show_a_notification_badge_when_its_value_is_0|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__is_selectable_when_not_selected|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__already_selected__is_not_selectable|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__already_selected__pops_to_root|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__animates_popping|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__when_presenting_a_root_view_controller__when_already_on_the_selected_tab__does_not_change_tab|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__when_presenting_a_root_view_controller__when_not_on_the_selected_tab__changes_tabs_in_an_animated_fashion|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__concerning_badges__shows_a_notification_badge|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__concerning_badges__updates_the_badge_count_in_the_data_source|
ARTopMenuViewControllerSpec | test_navigation__favorites__logged_in__tab_behavior__concerning_badges__does_not_show_a_notification_badge_when_its_value_is_0|
ARTopMenuViewControllerSpec | test_navigation__routing__supports_routing_to_paths|
UIViewController_TopMenuViewControllerSpec | test_returns_a_TopMenuVC_for_view_controllers_that_are_directly_in_a_heirarchy_with_a_ARTopMenuVC|
UIViewController_TopMenuViewControllerSpec | test_returns_a_TopMenuVC_for_view_controllers_that_are_deeper_in_a_heirarchy_with_a_ARTopMenuVC|


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by orta** [comment]
@sarahscott this is green now

File: Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift [hunk 7]
```diff
@@ -15,6 +15,7 @@ extension RefinementOptionsViewController {
         cancelButton.alignTopEdgeWithView(view, predicate: "10")
         cancelButton.alignTrailingEdgeWithView(view, predicate: "-10")
 
+        // This isn't normally running inside a nav, so needs to create it's own
```
Comment (sarahscott) [inline Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift]: it's -> its

File: Artsy/Networking/ArtworkSources/ARGeneArtworksNetworkModel.m [hunk 8]
```diff
@@ -31,13 +31,13 @@ - (id)initWithGeneID:(NSString *)geneID
     return [self initWithGene:[[Gene alloc] initWithGeneID:geneID]];
 }
 
-- (void)getGene:(void (^)(GeneViewModel *viewModel))success
+- (void)getGene:(void (^)(Gene *viewModel))success
```
Comment (sarahscott) [inline Artsy/Networking/ArtworkSources/ARGeneArtworksNetworkModel.m]: I'm totally ok with calling this class `Gene` instead of `GeneViewModel` but I'm curious what the motivation is behind the name change. I personally would assume that a class called `Gene` is more of a model than a view model.

File: Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift [hunk 9]
```diff
@@ -60,38 +78,59 @@ private extension RefinementOptionsViewController {
...
             $0.delegate = tableViewHandler
 
-            let tableViewHeight = 44 * currentSettings.numberOfRowsInSection(0) - 1 // -1 to cut off the bottom-most separator that we'll manually add below.
-            $0.constrainHeight("\(tableViewHeight)")
+            let combinedRowCount = (0..<currentSettings.numberOfSections).map(currentSettings.numberOfRowsInSection).reduce(0, combine:+)
+            let combinedTitleCount = Int(TableViewTitleHeight) * currentSettings.numberOfSections
+
+            let tableViewHeight = combinedTitleCount + (44 * combinedRowCount) + Int(bottomHeight)
+
+            // Only allow scrolling for large refine settings
```
Comment (sarahscott) [inline Artsy/View_Controllers/Auction/RefinementOptionsViewController+Private.swift]: These comments are 🔥

File: Artsy/Networking/ArtworkSources/ARGeneArtworksNetworkModel.m (hunk 8 above)
Comment (sarahscott) [inline Artsy/Networking/ArtworkSources/ARGeneArtworksNetworkModel.m]: Or this could be clearer by naming the variable `gene` instead of `viewModel`.

Comment (orta) [inline Artsy/Networking/ArtworkSources/ARGeneArtworksNetworkModel.m]: I removed the `GeneViewModel` so the variable should be called `gene` agreed, will do so in another PR 

That said, this entire thing should get deleted soon

**Comment by orta** [comment]
Pushed changes, will merge on green
